aptos-move/framework/aptos-token/sources/property_map.move

```diff
@@ -160,6 +162,35 @@ module aptos_token::property_map {
         *property_val = value;
     }
 
+    public fun create_property_value_raw(
+        value: vector<u8>,
+        type: String
+    ): PropertyValue {
+        PropertyValue {
+            value,
+            type,
+        }
+    }
+
+    /// create a property value from generic type data
+    public fun create_property_value<T: copy>(data: &T): PropertyValue {
+        if (type_name<T>() == string::utf8(b"bool")) {
+            create_property_value_raw(bcs::to_bytes<T>(data), string::utf8(b"bool"))
```

Comment:
Can we pass type_name<T>() directly to create_property_value_raw to avoid duplication?